2_process/src/lake_sequences_mntoha.py

```diff
@@ -0,0 +1,202 @@
+        obs_all = obs_all.join(ice_flags, how='left')
+
+        # Get attributes
+        obs_all['area'] = lake['area']
+        obs_all['lon'] = lake['centroid_lon']
+        obs_all['lat'] = lake['centroid_lat']
+        obs_all['elevation'] = lake['elevation']
+        # Now obs_all is one long timeseries with depth-specific observations,
+        # drivers, clarity, ice_flags, and attributes as columns
+
+        # Split into segments
+
+        # Let's make some rules for segments:
+        # 1. sequences are sequence_length days long
+        # 2. First spinup_time days are spinup, so don't have observations then
+        # 3. Each next sequence starts sequence_offset days later than the previous
+        # 4. Discard sequences that lack observations
+        # 5. We know going in that the first and last sequence have observations
+
+        # Convert dataframe to numpy array
+        obs_all_array = obs_all.to_numpy(dtype=np.float32)
+        # Create a strided view into obs_all_array
+        # This creates a numpy object with sequences of length sequence_length
+        # and offsets of sequence_offset without making a copy the array
+        all_sequences = np.lib.stride_tricks.sliding_window_view(
+            obs_all_array, sequence_length, axis=0
+        )[::sequence_offset, :]
+        # shape is now (# sequences, # features + depths, sequence_length)
+        # The first `len(depths)` elements in the middle dimension are observations
+
+        # Mask sequences that have only NaN observations
+        mask_nan_sequences = np.any(np.any(~np.isnan(all_sequences[:, :len(depths), :]), axis=2), axis=1)
+        # sequences to keep
+        lake_sequences = all_sequences[mask_nan_sequences, :, :].transpose((0, 2, 1))
+        # Now, shape is (# sequences, sequence_length, # features + depths)
+
+        # also keep the last `sequence_length` days, since the final observation is always non-NaN,
+        # unless all_sequences already has that last sequence 
+        if (obs_all_array.shape[0] - sequence_length) % sequence_offset != 0:
+            lake_sequences = np.concatenate((lake_sequences, np.array([obs_all_array[-sequence_length:, :]])), axis=0)
+
+    return lake_sequences
+
+
+def main(site_id,
+         out_file,
+         metadata_augmented_file,
+         obs_interpolated_file,
+         drivers_file,
+         clarity_file,
+         ice_flags_file,
+         config):
+    """
+    Process raw data for one MNTOHA lake
+    Save those data to a numpy file
+
+    :param site_id: site_id of lake to process
+    :param out_file: Filename of npy file to save
+    :param metadata_augmented_file: lake metadata file, augmented with elevation
+    :param obs_interpolated_file: temperature observations file, interpolated
+        to LSTM depths
+    :param drivers_file: Meteorological drivers csv for this lake
+    :param clarity_file: Time-varying lake clarity csv
```

Comment:
The clarity file is also 'for this lake', correct?

Reply:
Correct! I changed the text.